src/com/codeenginestudio/bookManagement/servlets/AddBookServlet.java

```diff
@@ -0,0 +1,50 @@
+package com.codeenginestudio.bookManagement.servlets;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.codeenginestudio.bookManagement.manage.ManageBook;
+import com.codeenginestudio.bookManagement.models.Book;
+import com.codeenginestudio.bookManagement.util.BookUtil;
+import com.codeenginestudio.bookManagement.validator.BookValidator;
+
+@WebServlet("/AddBookServlet")
+public class AddBookServlet extends HttpServlet {
+
+	protected void doGet(HttpServletRequest request, HttpServletResponse response)
+			throws ServletException, IOException {
+
+		BookUtil._displayView(request, response, "/view/AddOrEditBook.jsp");
+	}
+
+	protected void doPost(HttpServletRequest request, HttpServletResponse response)
+			throws ServletException, IOException {
+
+		String bookName = request.getParameter("bookName");
+		String bookAuthor = request.getParameter("bookAuthor");
```

Comment:
IMO, we should create a new book first. After that, validate with that book. Fail -> Return to edit page. Valid -> We don't need to create another one!